pkg/network/tracer/tracer_classification_test.go

```diff
@@ -1750,6 +1752,61 @@ func testEdgeCasesProtocolClassification(t *testing.T, tr *Tracer, clientHost, t
 			teardown:   teardown,
 			validation: validateProtocolConnection(&protocols.Stack{Application: protocols.HTTP}),
 		},
+		{
+			// This test checks than we are not classification a connection as
+			// carrying gRPC traffic without a prior classification as HTTP2.
+			name: "GRPC without prior HTTP2 classification",
```

Comment:
To make that test reliable, you also need to add the happy flow, so we can know, that only upon missing http2 classification, we're not making GRPC classification